src/main/java/nextstep/optional/Users.java

```diff
@@ -12,12 +12,21 @@
             new User("jk", 40),
             new User("honux", 45));
 
+//    User getUser(String name) {
+//        for (User user : users) {
+//            if (user.matchName(name)) {
+//                return user;
+//            }
+//        }
+//        return DEFAULT_USER;
+//    }
+
+    /**
+     * stream 과 Optional 을 사용한 리팩토링
+     */
     User getUser(String name) {
-        for (User user : users) {
-            if (user.matchName(name)) {
-                return user;
-            }
-        }
-        return DEFAULT_USER;
+        return users.stream()
+                .filter((user) -> user.getName().equals(name))
```

Comment:
```suggestion
                .filter((user) -> user.matchName(name))
```

user 객체가 가진 matchName 메소드를 사용해봐도 좋을 것 같네요.